src/Codeception/Codecept.php

```diff
@@ -57,12 +57,14 @@ class Codecept
     public function __construct($options = array()) {
         $this->result = new \PHPUnit_Framework_TestResult;
         $this->dispatcher = new EventDispatcher();
-        $this->loadExtensions($this->options);
+
+        $baseOptions = $this->mergeOptions($options);
+
+        $this->loadExtensions($baseOptions);
 
         $this->config = Configuration::config();
 
-        $this->options = array_merge($this->options, $this->config['settings']);
-        $this->options = array_merge($this->options, $options);
+        $this->options = $this->mergeOptions($options);
```

Comment:
is there a reason why `mergeOptions` is executed 2 times on the same data?
can merged options be reused?